tools/mcsema_disass/ida7/segment.py

```diff
@@ -335,7 +335,15 @@ def process_segments(binary_is_pie):
   """Pre-process a segment and try to fill in as many cross-references
   as is possible."""
 
-  seg_eas = [ea for ea in idautils.Segments() if not is_invalid_ea(ea)]
+  # NOTE(artem): IDA7 will add "LOAD" segments for parts of the program
+  # loaded into memory but not defined in a section. Ignore these since
+  # for normal compiler generated applications they provide no benefit
+  # but add lots of extra noise
+  def bad_seg(ea):
```

Comment:
Maybe make it a real function `is_bad_segment` instead of being a nested function. It's not obvious that nesting helps here...

Reply:
My original reason for nesting was to make it local to usage since its not used anywhere else

Reply:
If there's a good chance its useful outside of this scope, i can totally see expanding it out and maybe giving it a more descriptive name :)